x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -3548,6 +3559,34 @@ export class RulesClient {
         })
       );
     }
+
+    // check for actions using frequency params if the rule has global frequency params defined
+    if (usesGlobalFreqParams) {
+      const actionsWithFrequency = actions.filter((action) => Boolean(action.frequency));
+      if (actionsWithFrequency.length) {
+        throw Boom.badRequest(
+          i18n.translate('xpack.alerting.rulesClient.validateActions.mixAndMatchFreqParams', {
+            defaultMessage:
+              'Cannot mix and match per-action frequency params when notify_when or throttle are globally defined: {groups}',
+            values: {
+              groups: actionsWithFrequency.map((a) => a.group).join(', '),
+            },
+          })
+        );
+      }
+    } else {
+      const actionsWithoutFrequency = actions.filter((action) => !action.frequency);
+      if (actionsWithoutFrequency.length) {
```

Comment:
Like that we know that one of the action is missing this attribute
```suggestion
      if (actionsWithoutFrequency.length !== actions.length) {
```

Reply:
If I commit this suggestion, it will **NOT** throw an error if:

- There is no global `notify_when` defined
- There are also no `frequency` parameters defined

I don't think we want that

Reply:
Yes you are right!